crates/red_knot_python_semantic/resources/mdtest/attributes.md

```diff
@@ -41,13 +46,17 @@ reveal_type(c_instance.declared_and_bound)  # revealed: bool
 # mypy and pyright do not show an error here.
 reveal_type(c_instance.possibly_undeclared_unbound)  # revealed: str
 
+reveal_type(c_instance.inferred_from_redefined_param)  # revealed: Unknown | None | int
```

Comment:
Thank you for adding this test case. I think we could probably move it directly below the `reveal_type(c_instance.inferred_from_param)` above (and similarly move the definition in `__init__` upwards). And add a small comment that explains why we union with `Unknown` in that case.